### PR TITLE
feat(connected-itinerary-body): make alert collapse configurable

### PR DESCRIPTION
**Pull request by amy-corson-ibigroup**
**Description:**
Use `alwaysCollapseAlerts` prop in connected itinerary body to configure whether users can collapse alerts.

**PR Checklist:**
- [ ] Does the code follow accessibility standards (WCAG 2.1 AA Compliant)?
- [ ] Are all languages supported (Internationalization/Localization)?
- [ ] Are appropriate Typescript types implemented?

### CODE REVIEW

File: example-config.yml [hunk 1]
```diff
@@ -347,6 +347,8 @@ itinerary:
   renderRouteNamesInBlocks: true
   # Whether the mode icons should be colored as well
   fillModeIcons: true
+  # Allow user to collapse alerts in itinerary body
+  collapseAlerts: true
```
Comment (miles-grant-ibigroup) [inline example-config.yml:351]: This is misleadingly named. Can we name it `allowUserAlertCollapsing`?

**Review by miles-grant-ibigroup — CHANGES_REQUESTED** [review]
One small nitpick!

File: lib/components/narrative/line-itin/connected-itinerary-body.js [hunk 2]
```diff
@@ -116,6 +117,7 @@ class ConnectedItineraryBody extends Component {
       <ItineraryBodyContainer>
         <StyledItineraryBody
           accessibilityScoreGradationMap={accessibilityScoreGradationMap}
+          alwaysCollapseAlerts={allowUserAlertCollapsing}
```
Comment (miles-grant-ibigroup) [inline lib/components/narrative/line-itin/connected-itinerary-body.js:120]:
```suggestion
          alwaysCollapseAlerts={!allowUserAlertCollapsing}
```
Should this be flipped?

**Review by miles-grant-ibigroup — APPROVED** [review]
Looks good! Just one boolean thing to deal with

File: lib/components/narrative/line-itin/connected-itinerary-body.js (hunk 2 above)
Comment (amy-corson-ibigroup) [reply lib/components/narrative/line-itin/connected-itinerary-body.js:120]: I don't think so, I'm setting `allowUserAlertCollapsing` to `true` in the config!

Comment (amy-corson-ibigroup) [reply lib/components/narrative/line-itin/connected-itinerary-body.js:120]: Unless we want alerts to collapse by default, which might be preferable?

Comment (miles-grant-ibigroup) [reply lib/components/narrative/line-itin/connected-itinerary-body.js:120]: But as it is right now if `allowUserAlertCollapsing` is true, then they won't be allowed to collapse them right? Or am I forgetting how the behavior works

Comment (amy-corson-ibigroup) [reply lib/components/narrative/line-itin/connected-itinerary-body.js:120]: No, `alwaysCollapseAlerts` needs to be true for the alerts to collapse

**Review by binh-dam-ibigroup — CHANGES_REQUESTED** [review]
Please add the new attribute to the corresponding config type.

File: lib/util/config-types.ts [hunk 3]
```diff
@@ -243,6 +243,7 @@ export interface ItineraryCostWeights {
 }
 
 export interface ItineraryConfig {
+  allowUserAlertCollapsing: boolean
```
Comment (binh-dam-ibigroup) [inline lib/util/config-types.ts:246]: Yes, but make it optional:
```suggestion
  allowUserAlertCollapsing?: boolean
```

**Review by binh-dam-ibigroup — APPROVED** [review]
Approving subject to https://github.com/opentripplanner/otp-react-redux/pull/1182/files#r1543097570.

**Pull request merged**